dipy/workflows/nn.py

```diff
@@ -72,3 +76,92 @@ def run(
             )
 
         return io_it
+
+
+class BiasFieldCorrectionFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return "bias_field_correction"
+
+    def run(
+        self,
+        input_files,
+        bval=None,
+        bvec=None,
+        method="n4",
+        threshold=0.5,
+        verbose=False,
+        out_dir="",
+        out_corrected="biasfield_corrected.nii.gz",
+    ):
+        """Correct bias field.
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bval : string, optional
+            Path to the b-value file.
+        bvec : string, optional
+            Path to the b-vector file.
+        method : string, optional
+            Bias field correction method. Choose from:
+                - 'n4': DeepN4 bias field correction.
+                  See :footcite:p:`Kanakaraj2024` for more details.
+                - 'b0': B0 bias field correction via normalization.
+
+            'n4' method is recommended for T1-weighted images where 'b0' method
+            is recommended for diffusion-weighted images.
+        threshold : float, optional
+            Threshold for cleaning the final correction field in DeepN4 method.
+        verbose : bool, optional
+            Print verbose output.
+        out_dir : string, optional
+            Output directory.
+        out_corrected : string, optional
+            Name of the corrected volume to be saved.
+
+        References
+        ----------
+        .. footbibliography::
+
+        """
+        io_it = self.get_io_iterator()
+
+        if method.lower() not in ["n4", "b0"]:
+            logging.error("Unknown bias field correction method. Choose from 'n4, b0'.")
+            sys.exit(1)
+
+        # TODO: update output file names. #3477 need to be merged first
+        for fpath, corrected_out_path in io_it:
+            logging.info(f"Applying bias field correction on {fpath}")
+
+            data, affine, img, voxsize = load_nifti(
+                fpath, return_img=True, return_voxsize=True
+            )
+
+            corrected_data = None
+            if method.lower() == "b0":
+                bvals, bvecs = read_bvals_bvecs(bval, bvec)
+                gtab = gradient_table(bvals, bvecs=bvecs)
+                b0 = extract_b0(data, gtab.b0s_mask)
+                for i in range(data.shape[-1]):
+                    data[..., i] = np.divide(
+                        data[..., i],
+                        b0,
+                        out=np.zeros_like(data[..., i]).astype(float),
+                        where=b0 != 0,
+                    )
+                corrected_data = data
+            elif method.lower() == "n4":
+                deepn4_model = DeepN4()
```

Comment:
@pjsjongsung I was not able to run the model with 4GB of graphics. You have a variable which allows to run on "cpu". It is determined in the init. https://github.com/dipy/dipy/blob/32ca8ada4c1ff0f1ec016b023420929ae6a16515/dipy/nn/torch/deepn4.py#L199

@skoudoro we can provide a device variable which can be set in the case torch can be provided in the workflow.

@pjsjongsung Please share your thoughts. Also this can be done in separate PR. 



@skoudoro we can provide a

Reply:
Yes we can add an argument that lets us choose the device. We can have something like `use_cuda` which will use cuda if available. I'll create a quick PR tomorrow

Reply:
added, thanks @maharshi-gor